tests/e2e_test/utils_test.go

```diff
@@ -202,3 +230,108 @@ func GetKymaStateMetricCount(ctx context.Context, kymaName, state string) (int,
 
 	return 0, nil
 }
+
+func GetManifestObjectKey(ctx context.Context, k8sClient client.Client, kymaName, templateName string) (
+	types.NamespacedName, error,
+) {
+	manifests := &v1beta2.ManifestList{}
+	if err := k8sClient.List(ctx, manifests); err != nil {
+		return types.NamespacedName{Namespace: "", Name: ""}, fmt.Errorf("manifest fetching failed: %w", err)
+	}
+	for _, m := range manifests.Items {
+		module := m
+		if ownerKyma, exists := module.Labels["operator.kyma-project.io/kyma-name"]; exists && ownerKyma == kymaName {
+			if strings.Contains(module.Name, templateName) {
+				return client.ObjectKeyFromObject(&module), nil
+			}
+		}
+	}
+	return types.NamespacedName{Namespace: "", Name: ""},
+		fmt.Errorf("manifest fetching failed: %w", errNoManifestFound)
+}
+
+func GetResourceObjectKey(ctx context.Context, k8sClient client.Client, manifestObjectKey types.NamespacedName) (
+	types.NamespacedName, error,
+) {
+	manifest := &v1beta2.Manifest{}
+	if err := k8sClient.Get(ctx, manifestObjectKey, manifest); err != nil {
+		return types.NamespacedName{Namespace: "", Name: ""}, err
+	}
+	name, found, err := unstructured.NestedString(manifest.Spec.Resource.Object, "metadata", "name")
+	if err != nil || !found {
+		return types.NamespacedName{Namespace: "", Name: ""}, errResourceParseFromManifest
+	}
+	namespace, found, err := unstructured.NestedString(manifest.Spec.Resource.Object, "metadata", "namespace")
+	if err != nil || !found {
+		return types.NamespacedName{Namespace: "", Name: ""}, errResourceParseFromManifest
+	}
+	return types.NamespacedName{Namespace: namespace, Name: name}, err
+}
+
+func AddFinalizerToSampleResource(ctx context.Context,
+	objKey types.NamespacedName,
+	k8sClient client.Client,
+	finalizer string,
+) error {
+	resource := &templateOperator.Sample{}
```

Comment:
you don't need to use this typed `&templateOperator.Sample{}`, instead, use `Unstructured{}`, then you can avoid introducing template operator as a dependency.